### ISSUE TITLE
aws_ses: Provide Access to DNS Record for Custom "From" Domain

### ISSUE THREAD

**Issue by akefirad** [aws/aws-cdk#23377]
### Describe the feature

Similar to `dkimRecords`, it'd be nice to have access to the records need for verification of `mailFromDomain`.

### Use Case

To use them in the output.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.53.0

### Environment details (OS name and version, etc.)

macOS 2.53.0

**Comment by peterwoodworth** [comment]
Can you add some more detail in terms of what constructs will need this, how this functionality might be achieved, and/or exactly what your use case would be?

**Comment by akefirad** [comment]
Sure, so I've got this:
```typescript
      const mainIdentity = new ses.EmailIdentity(this, "MainIdentity", {
        identity: ses.Identity.domain("example.com"),
        mailFromDomain: "mail.example.com",
      });

      mainIdentity.dkimRecords.forEach(
        (r, i) =>
          new cdk.CfnOutput(this, `MainIdentityEasyDkimRecord${i}`, {
            value: r.name + " CNAME " + r.value,
            description: `DKIM Record #${i}`,
          }),
      );

      // FIXME: how to get the (two: MX and TXT) records of mail-from-domain thingy?
```
Now as you can see, I can easily *output* the DNS records for DKIM. But I'm not sure how to get the required DNS records for `mailFromDomain ` (two records, one `MX` and one `TXT`):
```
MX     mail.example.com     10 feedback-smtp.eu-central-1.amazonses.com
MX     mail.example.com        "v=spf1 include:amazonses.com ~all"
```
Now that I look at this again, it seems to be the same (per region). Does that make sense?

**Comment by peterwoodworth** [comment]
I see, thanks for the elaboration 🙂 

We are creating these resources in the constructor here https://github.com/aws/aws-cdk/blob/30f1ae4bb03015bb14b80eda619dc6a158722562/packages/%40aws-cdk/aws-ses/lib/email-identity.ts#L405-L420

These recordsets aren't being exposed as props of the EmailIdentity construct, so they're a bit tricky to access. You can do so by using [escape hatches](https://docs.aws.amazon.com/cdk/v2/guide/cfn_layer.html) to access these constructs. You should be able to find each of these as children of the `EmailIdentity` construct by searching for the ID you see in the code above. Once you have accessed these resources in your CDK code, you should be able to return their domain names.

As a feature request, we can certainly expose these resources to make them easier to access. I've marked this issue as p2, which means that we are unable to work on this immediately.

Check out our [contributing guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) if you're interested in contributing yourself - there's a low chance the team will be able to address this soon but we can try to review a PR 🙂

**Comment by akefirad** [comment]
Right, makes sense. Considering the fact that it’s almost static data, one can hard code it in their code. So I’d agree with the deprioritization. Feel free to close the issue.

**Comment by khushail** [comment]
> Right, makes sense. Considering the fact that it’s almost static data, one can hard code it in their code. So I’d agree with the deprioritization. Feel free to close the issue.

closing the issue as suggested

**Comment by peterwoodworth** [comment]
@khushail I suggest keeping this open as a feature request. If nothing has changed since I last commented, this could be made easier for users either through proper documentation of my guidance, or through directly exposing the children as props on the construct for easier access
